src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -638,7 +638,16 @@ TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extent(
 
   assert(extent->is_valid() && !extent->is_initial_pending());
   if (extent->is_dirty()) {
+    assert(extent->get_version() > 0);
+    if (extent->get_version() == 1 && extent->is_mutation_pending()) {
+      t.get_rewrite_stats().account_n_dirty();
+    } else {
+      t.get_rewrite_stats().account_dirty(extent->get_version());
+    }
```

Comment:
https://github.com/ceph/ceph/pull/59391#issuecomment-2303613403

Reply:
> Our test didn't include the version 0 extents.

This PR also excludes version 1 -- mutation-pending extents.

> What's more, we counted the accumulated modifications by different extent types.

This PR categorized the extents into 3 major types (data, mdat and phys), to make the report look less verbose.

> the update of deltas metric is at seastore_types.h:1212, empty deltas count is excluded.

This PR only accounts version instead of the number of deltas since clean.

@zhscn @xxhdx1985126 Does this look useful to have a rough understanding of whether trimming is too frequently ?

Reply:
> > Our test didn't include the version 0 extents.
> 
> This PR also excludes version 1 -- mutation-pending extents.

I think that dirty extents can also have only 1 version, right? So, this pr excludes mutation-pending extents but not all one-version extents, am I right?

> This PR only accounts version instead of the number of deltas since clean.

IIUC, a dirty extent's version should be the same as the number of deltas it absorbed, right?

> @zhscn @xxhdx1985126 Does this look useful to have a rough understanding of whether trimming is too frequently ?

Yeah, of course:-)

Reply:
> I think that dirty extents can also have only 1 version, right? So, this pr excludes mutation-pending extents but not all one-version extents, am I right?

Correct, dirty extent which version is 1 are accounted.

> IIUC, a dirty extent's version should be the same as the number of deltas it absorbed, right?

I think not, the version only increments upon `duplicate_for_write()`, which may involve multiple deltas.

Reply:
> I think not, the version only increments upon `duplicate_for_write()`, which may involve multiple deltas.

Ah, sorry, I didn't make myself clear. By the number of delat, I actually meant the number of `CachedExtent::get_delta()` invocations happened. IIUC, each `duplicate_for_write()` should corresponds to only one `get_delta` which is the one called in `Cache::prepare_record` when iterating through the `mutated_block_list`. Am I right?

Reply:
Yeah, I think that's correct.